### ISSUE TITLE
ms-swift-3.0微调后推理结果与ms-swift-2.x版本相差较大。

### ISSUE THREAD

**Issue by Betty-J** [modelscope/ms-swift#2788]
您好，我使用相同的数据集微调同一多模态模型，使用旧版 ms-swift 时可以得到比较理想的结果，更新到 ms-swift-3.0后微调无任何报错，但评测结果会比旧版差很多，以下是我的运行命令：

SIZE_FACTOR=8 MAX_PIXELS=602112 CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 NPROC_PER_NODE=8 swift sft \
    --tuner_backend peft \
    --model /Qwen2_VL/Qwen2-VL-2B-Instruct \
    --train_type lora \
    --seed 42 \
    --torch_dtype bfloat16 \
    --lora_dtype bfloat16 \
    --output_dir /Qwen2-VL-2B-Instruct \
    --dataset train.jsonl \
    --val_dataset val.jsonl \
    --num_train_epochs 60 \
    --max_length 2048 \
    --system 'you are a helpful assistant. ' \
    --lora_rank 16 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --freeze_llm false \
    --freeze_vit false \
    --freeze_aligner false \
    --gradient_checkpointing true \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 2 \
    --gradient_accumulation_steps 1 \
    --warmup_ratio 0.05 \
    --eval_steps 200 \
    --save_steps 200 \
    --save_total_limit 3 \
    --logging_steps 5 \
    --learning_rate 4e-5 \
    --attn_impl sdpa \
    --deepspeed zero2 \

版本：
ms-swift                          3.0.0.dev0 
transformers                   4.47.1

推理后结果存在的主要问题：1. 重复回答某句话直到输出token达到上限；2. 不按照给定的训练集格式回答；3. 回答出现无语义信息的乱码或符号。试着更改过 --freeze_vit、freeze_aligner 但影响不大。

尝试了多个不同的 VLM model，普遍存在上述问题。

**请问我遇到的问题可能由什么原因导致的呢，以及我的微调参数设置是否有问题。**

**Comment by mztknhao1** [comment]
我也遇到同样的问题，我实在昇腾卡上做的训练

**Comment by Jintao-Huang** [comment]
拉一下最新的代码试试

**Comment by Jintao-Huang** [comment]
参考这里：https://github.com/modelscope/ms-swift/blob/main/examples/train/multimodal/ocr.sh
